src/axios.js

```diff
@@ -0,0 +1,4 @@
+import axios from 'axios';
```

Comment:
A modular approach would be better, create an api folder in src, where we will have 3 items
- api configuration -> (successInterceptor, errorInterceptor, ,requestInterceptor)
`import axios from 'axios';
import config from '@/api/config';
import authService from '@/api/authService'

axios.defaults.baseURL = config.API_HOST;
export const statusCodesToHandle = [400, 401, 403, 404, 500];

export function successInterceptor(response) {
    return response.data;
}

export async function errorInterceptor(error) {
    const { status } = error.response;
    let errors = '';

    if (statusCodesToHandle.includes(status)) {
        errors = mapErrors(error.response.data);
    }

    error.errors = errors;

    return Promise.reject(error);
}

export function requestInterceptor(config) {
    const token = authService.getToken();

    if (!config.headers['Authorization'] && token) {
        config.headers['Authorization'] = token;
    }

    return config;
}

function mapErrors(data) {
    let errors = '';

    if (typeof data === 'string') {
        return data
    }
    if (!data.errors && data.message) {
        return data.message
    }
    if (Array.isArray(data.errors)) {
        const hasStringErrors = typeof data.errors[0] === 'string'

        if (hasStringErrors) {
            return data.errors[0]
        }

        data.errors = data.errors[0]
    }
    for (let err in data.errors) {
        let errorRow = data.errors[err].join('</br>');
        if (!errorRow.endsWith('</br>')) {
            errorRow += '</br>';
        }
        errors += errorRow;
    }
    return errors;
}

axios.interceptors.response.use(successInterceptor, errorInterceptor);
axios.interceptors.request.use(requestInterceptor);
`
- config -> here we will have the BE endpoint `let API_HOST = 'https://api-internship.binarcode.com';

if (process.env.VUE_APP_API_URL) {
    API_HOST = process.env.VUE_APP_API_URL
}

export default {
    API_HOST: API_HOST,
};`
- authentification service -> here we will have a folder for every type of BE action, auth, blogs, profile etc
`import axios from 'axios';

export default {
    setToken(token) {
        localStorage.setItem('token', token);
    },
    getToken() {
        return localStorage.getItem('token');
    },
    async register(data) {
        return await axios.post('/api/register', data);
    },
}`

Reply:
Let me know if it works